### PR TITLE
Documentation ConfigCollection and ResolutionHelper 

### PR DESCRIPTION
**Pull request by jakejohns**
First attempt to address #155 

Input greatly appreciated.
Especially in regards to the where/how about the `ResolutionHelper` as noted in #143

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -64,3 +64,39 @@ $di = $container_builder->newConfiguredInstance([
     'Aura\Web\_Config\Common',
 ]);
 ```
+
+If you have a package which combines a number of disparate components that
+each provide a ContainerConfig you could bundle them together using the
```
Comment (harikt) [inline docs/config.md]: May be backticks for `ContainerConfig` . Same for all classes ?

**Review by harikt — APPROVED** [review]

**Comment by harikt** [comment]
@jakejohns do you think we can merge this PR as is ?

**Comment by jakejohns** [comment]
@harikt sorry I took so long! It's your call if its ready, I guess. Thanks for the reminder!

**Comment by harikt** [comment]
@jakejohns no issues. I was also carried away with some other stuffs.

**Pull request merged**